processor/processor.go

```diff
@@ -836,6 +840,18 @@ func (proc *Handle) backendConfigSubscriber(ctx context.Context) {
 	}
 }
 
+func ConvertMapToList(credentialsMap map[string]backendconfig.Credential) []backendconfig.Credential {
+	var credentialsList []backendconfig.Credential
+	for _, credential := range credentialsMap {
+		credentialsList = append(credentialsList, backendconfig.Credential{
+			Key:      credential.Key,
+			Value:    credential.Value,
+			IsSecret: credential.IsSecret,
+		})
+	}
+	return credentialsList
+}
```

Comment:
We are loosing the credential identifier in this process and only sending list of k/v combinations. Why are we doing that ?

We should contain necessary information like identifier for debugging purposes as well. Wdyt ?!